Source/Core/Duality/Utility/Pooling/QueuedPool.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Duality.Utility.Pooling
+{
+	internal class QueuedPool<T> where T : class, new()
+	{
+		private readonly Queue<T> pool = new Queue<T>();
+
+		public T GetOne()
+		{
+			T result = null;
+
+			if (this.pool.Count > 0)
+				result = this.pool.Dequeue();
+			else
+				result = new T();
+
+			if (result is IPoolable)
+				(result as IPoolable).OnPickup();
+
+			return result;
+		}
+
+		public void ReturnOne(T obj)
+		{
+			if (obj is IPoolable)
+				(obj as IPoolable).OnReturn();
```

Comment:
Recommend `as` then check for null.

Reply:
You can just do the is check nowadays and put the result in a variable: 'obj is IPoolable poolable'